core/providers/elixir/elixir.go

```diff
@@ -60,6 +62,44 @@ func (p *ElixirProvider) Plan(ctx *generate.GenerateContext) error {
 	})
 	ctx.Deploy.StartCmd = p.GetStartCommand(ctx)
 
+	// Node (if necessary)
+	if app, err := app.NewApp(ctx.App.Source + "/assets"); err == nil {
+		// All providers assume they're running in the application root
+		// but Phoenix puts it in the assets folder, so we have to lie to the provider
+		ctx.App = app
```

Comment:
For safety I wonder if we want an `originalApp` var that we can reset on the context e.g.

```go
originalApp := ctx.app
ctx.App = app
...
ctx.App = originalApp
```

Reply:
Otherwise looks good to me

Reply:
With the current codebase it doesn't matter since `ctx.App` is copied into places that use it, but you're right that it might not always be that way and the change should be cleaned up. I'll make that change, and move the logic into a function so I can use `defer` and reduce some of the nested `if`s.